components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -930,4 +934,32 @@ private String getTenantDomainFromUserName(AuthenticationContext context, String
         }
         return MultitenantUtils.getTenantDomain(username);
     }
+
+    private boolean isURLContainSensitiveData(HttpServletRequest request, HttpServletResponse response,
+                                              AuthenticationContext context) throws AuthenticationFailedException {
+
+        if (StringUtils.contains(request.getQueryString(), BasicAuthenticatorConstants.USER_NAME) &&
```

Comment:
```suggestion
        if (StringUtils.contains(request.getQueryString(), BasicAuthenticatorConstants.USER_NAME) ||
```

So if even one parameter is available, let's fail.